crates/ibc/src/core/ics03_connection/connection.rs

```diff
@@ -97,6 +109,125 @@ pub struct ConnectionEnd {
     delay_period: Duration,
 }
 
+mod sealed {
```

Comment:
This whole module can be removed by annotating `ConnectionEnd` directly (with `cfg_attr`), except for the `TypeInfo` implementation on line 199.

Reply:
> 

Oh, it doesn't work here, `borsh` and `partiy-scale-code` don't automatically implement Encode and Decode for Duration, you have to do it manually.

Reply:
Ah yes, you're right; however `parity_scale_codec::Encode` and `parity_scale_codec::Decode` can be derived automatically.

```diff
diff --git a/crates/ibc/src/core/ics03_connection/connection.rs b/crates/ibc/src/core/ics03_connection/connection.rs
index c8717659..a52aab28 100644
--- a/crates/ibc/src/core/ics03_connection/connection.rs
+++ b/crates/ibc/src/core/ics03_connection/connection.rs
@@ -100,6 +100,13 @@ impl From<IdentifiedConnectionEnd> for RawIdentifiedConnection {
     }
 }
 
+#[cfg_attr(
+    feature = "parity-scale-codec",
+    derive(
+        parity_scale_codec::Encode,
+        parity_scale_codec::Decode,
+    )
+)]
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub struct ConnectionEnd {
     pub state: State,
@@ -112,14 +119,6 @@ pub struct ConnectionEnd {
 mod sealed {
     use super::*;
 
-    #[cfg_attr(
-        feature = "parity-scale-codec",
-        derive(
-            parity_scale_codec::Encode,
-            parity_scale_codec::Decode,
-            scale_info::TypeInfo
-        )
-    )]
     #[cfg_attr(
         feature = "borsh",
         derive(borsh::BorshSerialize, borsh::BorshDeserialize)
@@ -177,24 +176,6 @@ mod sealed {
         }
     }
 
-    #[cfg(feature = "parity-scale-codec")]
-    impl parity_scale_codec::Encode for ConnectionEnd {
-        fn encode_to<T: parity_scale_codec::Output + ?Sized>(&self, writer: &mut T) {
-            let value = InnerConnectionEnd::from(self.clone());
-            value.encode_to(writer);
-        }
-    }
-
-    #[cfg(feature = "parity-scale-codec")]
-    impl parity_scale_codec::Decode for ConnectionEnd {
-        fn decode<I: parity_scale_codec::Input>(
-            input: &mut I,
-        ) -> Result<Self, parity_scale_codec::Error> {
-            let result = InnerConnectionEnd::decode(input)?;
-            Ok(ConnectionEnd::from(result))
-        }
-    }
-
     #[cfg(feature = "parity-scale-codec")]
     impl scale_info::TypeInfo for ConnectionEnd {
         type Identity = Self;
```

Reply:
> Ah yes, you're right; however parity_scale_codec::Encode and parity_scale_codec::Decode can be derived automatically.

Done